src/lib/utils/toStorageCurrency.ts

```diff
@@ -0,0 +1,34 @@
+import { exchangeRate } from '$lib/stores/exchangeRate';
+import { FRACTION_DIGITS_PER_CURRENCY, createPrice, type Currency } from '$lib/types/Currency';
+import { get } from 'svelte/store';
+
+/**
+ * Convert amount to target currency with storage precision (4 decimals for fiat, 8 for BTC).
+ * Use this when storing prices in database to preserve full precision for VAT calculations.
+ *
+ * For display purposes, use toCurrency() which rounds to FRACTION_DIGITS_PER_CURRENCY (2 decimals for fiat).
+ */
+export function toStorageCurrency(
+	targetCurrency: Currency,
+	amount: number,
+	fromCurrency: Currency
+): number {
```

Comment:
It's really unexpected for this function to return a scalar number: The number requires more context to be understood (at least currency). Perhaps it would be best for this to return a `Price` (with currency and precision)?